### PR TITLE
Add example and test for showing how to setup up single instance per component

### PR DESCRIPTION
**Pull request by aderington**
#128

### CODE REVIEW

File: sample/app/src/test/java/com/squareup/anvil/sample/RandomModuleTest.kt [hunk 1]
```diff
@@ -1,12 +1,18 @@
...
+import com.squareup.scopes.ComponentHolder
 import org.junit.Test
 
 class RandomModuleTest {
   @Test fun `module in this app module is contributed`() {
-    val component = DaggerAppComponent.create() as RandomComponent
+    ComponentHolder.components.add(DaggerAppComponent.create())
 
-    assertThat(component.string()).isEqualTo(RandomModule.provideString())
+    val component = ComponentHolder.component<RandomComponent>()
```
Comment (vRallev) [inline sample/app/src/test/java/com/squareup/anvil/sample/RandomModuleTest.kt:12]: Using the `ComponentHolder` for this test is redundant.

File: sample/scopes/build.gradle [hunk 2]
```diff
@@ -2,4 +2,5 @@ apply plugin: 'org.jetbrains.kotlin.jvm'
 
 dependencies {
   api deps.kotlin.stdlib
+  implementation "javax.inject:javax.inject:1"
```
Comment (vRallev) [inline sample/scopes/build.gradle:5]: That needs to be `api`.

**Review by vRallev — CHANGES_REQUESTED** [review]
Change looks good. Only the nits need to be changed.

File: sample/scopes/build.gradle [hunk 3]
```diff
@@ -2,4 +2,5 @@ apply plugin: 'org.jetbrains.kotlin.jvm'
 
 dependencies {
   api deps.kotlin.stdlib
+  api "javax.inject:javax.inject:1"
```
Comment (vRallev) [inline sample/scopes/build.gradle:5]: Ah sorry, I missed that in the first review. In this repository we list all dependencies in the `dependencies.gradle` file. You should do the same for this dependency and then use `api deps.inject` here.

**Review by vRallev — CHANGES_REQUESTED** [review]

File: sample/scopes/build.gradle (hunk 3 above)
Comment (aderington) [reply sample/scopes/build.gradle:5]: Sounds good, I'll make that update!

**Review by vRallev — APPROVED** [review]
Looks good. Please don't forget to sign the CLA. Otherwise I can't merge.

**Comment by aderington** [comment]
@vRallev What's the best link for information about signing the CLA?

**Comment by vRallev** [comment]
Scroll to the top: https://github.com/square/anvil/pull/129#issuecomment-697612994

**Comment by aderington** [comment]
Done!

**Pull request merged**